core/src/main/scala/org/apache/spark/sql/internal/SnappySessionState.scala

```diff
@@ -503,21 +503,31 @@ class SnappyConf(@transient val session: SnappySession)
     case Property.PlanCaching.name =>
       value match {
         case Some(boolval) => {
-          if (boolval.toString.toBoolean) session.clearPlanCache()
+          if (boolval.toString.toBoolean) {
+            session.clearPlanCache()
+          }
+          session.planCaching = boolval.toString.toBoolean
```

Comment:
can store boolVal.toString.toBoolean

Reply:
Thought but ignored. Hardly matters for this use case.